src/fetching/getSurveysWithElements.ts

```diff
@@ -0,0 +1,18 @@
+import { defaultFetch } from '.';
+import { ZetkinSurvey, ZetkinSurveyExtended } from '../types/zetkin';
+
+export default function getSurveys(orgId : string, fetch = defaultFetch) {
+    return async (): Promise<ZetkinSurveyExtended[]> => {
+
+        const res = await fetch(`/orgs/${orgId}/surveys`);
+        const body = await res.json();
+
+        const surveys: ZetkinSurvey[] = body.data;
+
+        const surveysWithElements = await Promise.all(surveys.map(async s =>
+            await fetch(`/orgs/${orgId}/surveys/${s.id}`)
+                .then(res => res.json())));
```

Comment:
Functions marked as `async` will be made by the runtime to return a `Promise`. So `async () => 5` returns a promise that resolves to `5` (rather than just returning 5) even though the function returns (almost) immediately because the actual logic and operations of it are not async.

The "old-school" pre-async/await way to write that same function is `() => Promise.resolve(5)`, i.e. return a promise that resolves to 5.

On these lines here, you have had to mark your map function as `async` because `Promise.all()` expects an array of promises, so the function needs to return a promise so that the resulting array is an array of promises.

But `fetch()` already returns a promise. So you shouldn't have to use an `async` function, just a function that returns whatever `fetch()` returns, i.e. this should work:

```tsx
const surveysWithElements = await Promise.all(surveys.map(s =>
  fetch(`/orgs/${orgId}/surveys/${s.id}`)
    .then(res => res.json())
  )
);
```

Note that it's not using `await` inside the map function. If you `await`, you get back the value _that the promise resolves to_, so the return value/type of the function becomes a survey. By not awaiting, the return value is a promise _that resolves to a survey_ which is what we want.